cmd/scollector/collectors/fastly.go

```diff
@@ -0,0 +1,374 @@
+	Status204          int64   `json:"status_204" div:"true" rate:"gauge" unit:"requests per second" desc:"The number of http requests with a 204 response code (No Content)."`
+	Status2xx          int64   `json:"status_2xx" div:"true" rate:"gauge" unit:"requests per second" desc:"The number of http requests with a 2xx response code (Success)."`
+	Status301          int64   `json:"status_301" div:"true" rate:"gauge" unit:"requests per second" desc:"The number of http requests with a 301 response code (Moved Permanently)."`
+	Status302          int64   `json:"status_302" div:"true" rate:"gauge" unit:"requests per second" desc:"The number of http requests with a 302 response code (Found)."`
+	Status304          int64   `json:"status_304" div:"true" rate:"gauge" unit:"requests per second" desc:"The number of http requests with a 304 response code."`
+	Status3xx          int64   `json:"status_3xx" div:"true" rate:"gauge" unit:"requests per second" desc:"The number of http requests with a 3xx response code (Redirection)."`
+	Status4xx          int64   `json:"status_4xx" div:"true" rate:"gauge" unit:"requests per second" desc:"The number of http requests with a 4xx response code (Client Error)."`
+	Status503          int64   `json:"status_503" div:"true" rate:"gauge" unit:"requests per second" desc:"The number of http requests with a 503 response code. (Service Unavailable)"`
+	Status5xx          int64   `json:"status_5xx" div:"true" rate:"gauge" unit:"requests per second" desc:"The number of http requests with a xxx response code."`
+	Synth              int64   `json:"synth"`
+	TLS                int64   `json:"tls"`
+	Uncacheable        int64   `json:"uncacheable" div:"true" rate:"gauge" unit:"requests per second" desc:"The number of http requests that were uncacheable."`
+	Video              int64   `json:"video"`
+}
+
+const fastlyDivDesc = " This metric is collected per minute, but we divide it by 60 seconds in order to normalize the rate to per second instead of per minute."
+
+func FastlyReflectAdd(md *opentsdb.MultiDataPoint, prefix, suffix string, st interface{}, timeStamp int64, ts opentsdb.TagSet) {
+	t := reflect.TypeOf(st)
+	valueOf := reflect.ValueOf(st)
+	for i := 0; i < t.NumField(); i++ {
+		field := t.Field(i)
+		value := valueOf.Field(i).Interface()
+		var (
+			jsonTag   = field.Tag.Get("json")
+			metricTag = field.Tag.Get("metric")
+			rateTag   = field.Tag.Get("rate")
+			unitTag   = field.Tag.Get("unit")
+			divTag    = field.Tag.Get("div")
+			descTag   = field.Tag.Get("desc")
+			exclude   = field.Tag.Get("exclude") != ""
+		)
+		if exclude || descTag == "" {
+			continue
+		}
+		metricName := jsonTag
+		if metricTag != "" {
+			metricName = metricTag
+		}
+		if metricName == "" {
+			slog.Errorf("Unable to determine metric name for field %s. Skipping.", field.Name)
+			continue
+		}
+		shouldDiv := divTag != ""
```

Comment:
Weak check which may cause bugs if things get extended. Why not check to see if it is == 'true'?